## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 419713`
Commit: `0849f6027ea957fe8789a4afb39437b822810c54`

---
### `x86_64-linux`
<details>
  <summary>:x: 24 packages failed to build:</summary>
  <ul>
    <li>archivy</li>
    <li>archivy.dist</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
    <li>newelle</li>
    <li>odoo</li>
    <li>odoo.dist</li>
    <li>odoo16</li>
    <li>odoo16.dist</li>
    <li>odoo17</li>
    <li>odoo17.dist</li>
    <li>open-web-calendar</li>
    <li>open-web-calendar.dist</li>
    <li>python312Packages.html-sanitizer</li>
    <li>python312Packages.html-sanitizer.dist</li>
    <li>python312Packages.trafilatura</li>
    <li>python312Packages.trafilatura.dist</li>
    <li>python313Packages.html-sanitizer</li>
    <li>python313Packages.html-sanitizer.dist</li>
    <li>python313Packages.trafilatura</li>
    <li>python313Packages.trafilatura.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 41 packages built:</summary>
  <ul>
    <li>changedetection-io</li>
    <li>changedetection-io.dist</li>
    <li>gnome-feeds</li>
    <li>offpunk</li>
    <li>offpunk.dist</li>
    <li>python312Packages.extruct</li>
    <li>python312Packages.extruct.dist</li>
    <li>python312Packages.html-text</li>
    <li>python312Packages.html-text.dist</li>
    <li>python312Packages.justext</li>
    <li>python312Packages.justext.dist</li>
    <li>python312Packages.lxml-html-clean</li>
    <li>python312Packages.lxml-html-clean.dist</li>
    <li>python312Packages.newspaper3k</li>
    <li>python312Packages.newspaper3k.dist</li>
    <li>python312Packages.pywebcopy</li>
    <li>python312Packages.pywebcopy.dist</li>
    <li>python312Packages.readability-lxml</li>
    <li>python312Packages.readability-lxml.dist</li>
    <li>python312Packages.recipe-scrapers</li>
    <li>python312Packages.recipe-scrapers.dist</li>
    <li>python313Packages.extruct</li>
    <li>python313Packages.extruct.dist</li>
    <li>python313Packages.html-text</li>
    <li>python313Packages.html-text.dist</li>
    <li>python313Packages.justext</li>
    <li>python313Packages.justext.dist</li>
    <li>python313Packages.lxml-html-clean</li>
    <li>python313Packages.lxml-html-clean.dist</li>
    <li>python313Packages.newspaper3k</li>
    <li>python313Packages.newspaper3k.dist</li>
    <li>python313Packages.pywebcopy</li>
    <li>python313Packages.pywebcopy.dist</li>
    <li>python313Packages.readability-lxml</li>
    <li>python313Packages.readability-lxml.dist</li>
    <li>python313Packages.recipe-scrapers</li>
    <li>python313Packages.recipe-scrapers.dist</li>
    <li>qutebrowser</li>
    <li>qutebrowser-qt5</li>
    <li>qutebrowser-qt5.dist</li>
    <li>qutebrowser.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 39 packages failed to build:</summary>
  <ul>
    <li>archivy</li>
    <li>archivy.dist</li>
    <li>changedetection-io</li>
    <li>changedetection-io.dist</li>
    <li>gnome-feeds</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
    <li>newelle</li>
    <li>odoo</li>
    <li>odoo.dist</li>
    <li>odoo16</li>
    <li>odoo16.dist</li>
    <li>odoo17</li>
    <li>odoo17.dist</li>
    <li>offpunk</li>
    <li>offpunk.dist</li>
    <li>open-web-calendar</li>
    <li>open-web-calendar.dist</li>
    <li>python312Packages.html-sanitizer</li>
    <li>python312Packages.html-sanitizer.dist</li>
    <li>python312Packages.recipe-scrapers</li>
    <li>python312Packages.recipe-scrapers.dist</li>
    <li>python312Packages.trafilatura</li>
    <li>python312Packages.trafilatura.dist</li>
    <li>python313Packages.extruct</li>
    <li>python313Packages.extruct.dist</li>
    <li>python313Packages.html-sanitizer</li>
    <li>python313Packages.html-sanitizer.dist</li>
    <li>python313Packages.recipe-scrapers</li>
    <li>python313Packages.recipe-scrapers.dist</li>
    <li>python313Packages.trafilatura</li>
    <li>python313Packages.trafilatura.dist</li>
    <li>qutebrowser</li>
    <li>qutebrowser-qt5</li>
    <li>qutebrowser-qt5.dist</li>
    <li>qutebrowser.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 26 packages built:</summary>
  <ul>
    <li>python312Packages.extruct</li>
    <li>python312Packages.extruct.dist</li>
    <li>python312Packages.html-text</li>
    <li>python312Packages.html-text.dist</li>
    <li>python312Packages.justext</li>
    <li>python312Packages.justext.dist</li>
    <li>python312Packages.lxml-html-clean</li>
    <li>python312Packages.lxml-html-clean.dist</li>
    <li>python312Packages.newspaper3k</li>
    <li>python312Packages.newspaper3k.dist</li>
    <li>python312Packages.pywebcopy</li>
    <li>python312Packages.pywebcopy.dist</li>
    <li>python312Packages.readability-lxml</li>
    <li>python312Packages.readability-lxml.dist</li>
    <li>python313Packages.html-text</li>
    <li>python313Packages.html-text.dist</li>
    <li>python313Packages.justext</li>
    <li>python313Packages.justext.dist</li>
    <li>python313Packages.lxml-html-clean</li>
    <li>python313Packages.lxml-html-clean.dist</li>
    <li>python313Packages.newspaper3k</li>
    <li>python313Packages.newspaper3k.dist</li>
    <li>python313Packages.pywebcopy</li>
    <li>python313Packages.pywebcopy.dist</li>
    <li>python313Packages.readability-lxml</li>
    <li>python313Packages.readability-lxml.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 55 packages failed to build:</summary>
  <ul>
    <li>archivy</li>
    <li>archivy.dist</li>
    <li>changedetection-io</li>
    <li>changedetection-io.dist</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
    <li>odoo</li>
    <li>odoo.dist</li>
    <li>odoo16</li>
    <li>odoo16.dist</li>
    <li>odoo17</li>
    <li>odoo17.dist</li>
    <li>offpunk</li>
    <li>offpunk.dist</li>
    <li>python312Packages.extruct</li>
    <li>python312Packages.extruct.dist</li>
    <li>python312Packages.html-sanitizer</li>
    <li>python312Packages.html-sanitizer.dist</li>
    <li>python312Packages.justext</li>
    <li>python312Packages.justext.dist</li>
    <li>python312Packages.newspaper3k</li>
    <li>python312Packages.newspaper3k.dist</li>
    <li>python312Packages.pywebcopy</li>
    <li>python312Packages.pywebcopy.dist</li>
    <li>python312Packages.readability-lxml</li>
    <li>python312Packages.readability-lxml.dist</li>
    <li>python312Packages.recipe-scrapers</li>
    <li>python312Packages.recipe-scrapers.dist</li>
    <li>python312Packages.trafilatura</li>
    <li>python312Packages.trafilatura.dist</li>
    <li>python313Packages.extruct</li>
    <li>python313Packages.extruct.dist</li>
    <li>python313Packages.html-sanitizer</li>
    <li>python313Packages.html-sanitizer.dist</li>
    <li>python313Packages.html-text</li>
    <li>python313Packages.html-text.dist</li>
    <li>python313Packages.justext</li>
    <li>python313Packages.justext.dist</li>
    <li>python313Packages.newspaper3k</li>
    <li>python313Packages.newspaper3k.dist</li>
    <li>python313Packages.pywebcopy</li>
    <li>python313Packages.pywebcopy.dist</li>
    <li>python313Packages.readability-lxml</li>
    <li>python313Packages.readability-lxml.dist</li>
    <li>python313Packages.recipe-scrapers</li>
    <li>python313Packages.recipe-scrapers.dist</li>
    <li>python313Packages.trafilatura</li>
    <li>python313Packages.trafilatura.dist</li>
    <li>qutebrowser</li>
    <li>qutebrowser-qt5</li>
    <li>qutebrowser-qt5.dist</li>
    <li>qutebrowser.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.html-text</li>
    <li>python312Packages.html-text.dist</li>
    <li>python312Packages.lxml-html-clean</li>
    <li>python312Packages.lxml-html-clean.dist</li>
    <li>python313Packages.lxml-html-clean</li>
    <li>python313Packages.lxml-html-clean.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 49 packages failed to build:</summary>
  <ul>
    <li>archivy</li>
    <li>archivy.dist</li>
    <li>changedetection-io</li>
    <li>changedetection-io.dist</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
    <li>offpunk</li>
    <li>offpunk.dist</li>
    <li>python312Packages.extruct</li>
    <li>python312Packages.extruct.dist</li>
    <li>python312Packages.html-sanitizer</li>
    <li>python312Packages.html-sanitizer.dist</li>
    <li>python312Packages.html-text</li>
    <li>python312Packages.html-text.dist</li>
    <li>python312Packages.justext</li>
    <li>python312Packages.justext.dist</li>
    <li>python312Packages.newspaper3k</li>
    <li>python312Packages.newspaper3k.dist</li>
    <li>python312Packages.pywebcopy</li>
    <li>python312Packages.pywebcopy.dist</li>
    <li>python312Packages.recipe-scrapers</li>
    <li>python312Packages.recipe-scrapers.dist</li>
    <li>python312Packages.trafilatura</li>
    <li>python312Packages.trafilatura.dist</li>
    <li>python313Packages.extruct</li>
    <li>python313Packages.extruct.dist</li>
    <li>python313Packages.html-sanitizer</li>
    <li>python313Packages.html-sanitizer.dist</li>
    <li>python313Packages.html-text</li>
    <li>python313Packages.html-text.dist</li>
    <li>python313Packages.justext</li>
    <li>python313Packages.justext.dist</li>
    <li>python313Packages.newspaper3k</li>
    <li>python313Packages.newspaper3k.dist</li>
    <li>python313Packages.pywebcopy</li>
    <li>python313Packages.pywebcopy.dist</li>
    <li>python313Packages.readability-lxml</li>
    <li>python313Packages.readability-lxml.dist</li>
    <li>python313Packages.recipe-scrapers</li>
    <li>python313Packages.recipe-scrapers.dist</li>
    <li>python313Packages.trafilatura</li>
    <li>python313Packages.trafilatura.dist</li>
    <li>qutebrowser</li>
    <li>qutebrowser-qt5</li>
    <li>qutebrowser-qt5.dist</li>
    <li>qutebrowser.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.lxml-html-clean</li>
    <li>python312Packages.lxml-html-clean.dist</li>
    <li>python312Packages.readability-lxml</li>
    <li>python312Packages.readability-lxml.dist</li>
    <li>python313Packages.lxml-html-clean</li>
    <li>python313Packages.lxml-html-clean.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>newelle</summary>
<pre>copying path '/nix/store/qq6sz80vzm6nfvvzf27sypvzzs65zxi6-python3.13-pip-install-test-0.5' from 'https://cache.nixos.org'...
copying path '/nix/store/080xadlmarmbsi497asbqhzvzk3y8p7a-python3.13-speechrecognition-3.14.3' from 'https://cache.nixos.org'...
Connection to build-box.nix-community.org closed by remote host.</pre>
</details>
<details>
<summary>odoo</summary>
<pre>copying odoo/addons/spreadsheet/tests/test_currency.py -> build/lib/odoo/addons/spreadsheet/tests
creating build/lib/odoo/addons/spreadsheet/utils
copying odoo/addons/spreadsheet/utils/__init__.py -> build/lib/odoo/addons/spreadsheet/utils
copying odoo/addons/spreadsheet/utils/validate_data.py -> build/lib/odoo/addons/spreadsheet/utils
copying odoo/addons/spreadsheet/utils/formatting.py -> build/lib/odoo/addons/spreadsheet/utils
creating build/lib/odoo/addons/payment_razorpay_oauth/models
copying odoo/addons/payment_razorpay_oauth/models/__init__.py -> build/lib/odoo/addons/payment_razorpay_oauth/models
copying odoo/addons/payment_razorpay_oauth/models/payment_provider.py -> build/lib/odoo/addons/payment_razorpay_oauth/models
creating build/lib/odoo/addons/payment_razorpay_oauth/controllers
copying odoo/addons/payment_razorpay_oauth/controllers/__init__.py -> build/lib/odoo/addons/payment_razorpay_oauth/controllers
copying odoo/addons/payment_razorpay_oauth/controllers/onboarding.py -> build/lib/odoo/addons/payment_razorpay_oauth/controllers
creating build/lib/odoo/addons/purchase_requisition_sale/tests
copying odoo/addons/purchase_requisition_sale/tests/test_purchase_requisition_sale.py -> build/lib/odoo/addons/purchase_requisition_sale/tests
copying odoo/addons/purchase_requisition_sale/tests/__init__.py -> build/lib/odoo/addons/purchase_requisition_sale/tests
creating build/lib/odoo/addons/purchase_requisition_sale/wizard
copying odoo/addons/purchase_requisition_sale/wizard/purchase_requisition_create_alternative.py -> build/lib/odoo/addons/purchase_requisition_sale/wizard
copying odoo/addons/purchase_requisition_sale/wizard/__init__.py -> build/lib/odoo/addons/purchase_requisition_sale/wizard
creating build/lib/odoo/addons/website_sale_mass_mailing/models
copying odoo/addons/website_sale_mass_mailing/models/website.py -> build/lib/odoo/addons/website_sale_mass_mailing/models
copying odoo/addo</pre>
</details>
<details>
<summary>odoo16</summary>
<pre>copying odoo/addons/l10n_in_edi_ewaybill/tests/test_edi_ewaybill_json.py -> build/lib/odoo/addons/l10n_in_edi_ewaybill/tests
creating build/lib/odoo/addons/l10n_in_edi_ewaybill/demo
copying odoo/addons/l10n_in_edi_ewaybill/demo/__init__.py -> build/lib/odoo/addons/l10n_in_edi_ewaybill/demo
copying odoo/addons/l10n_in_edi_ewaybill/demo/chart_template.py -> build/lib/odoo/addons/l10n_in_edi_ewaybill/demo
creating build/lib/odoo/addons/website_event_jitsi/models
copying odoo/addons/website_event_jitsi/models/__init__.py -> build/lib/odoo/addons/website_event_jitsi/models
copying odoo/addons/website_event_jitsi/models/res_config_settings.py -> build/lib/odoo/addons/website_event_jitsi/models
creating build/lib/odoo/addons/hw_drivers/controllers
copying odoo/addons/hw_drivers/controllers/driver.py -> build/lib/odoo/addons/hw_drivers/controllers
copying odoo/addons/hw_drivers/controllers/__init__.py -> build/lib/odoo/addons/hw_drivers/controllers
copying odoo/addons/hw_drivers/controllers/proxy.py -> build/lib/odoo/addons/hw_drivers/controllers
creating build/lib/odoo/addons/hr_expense/models
copying odoo/addons/hr_expense/models/ir_attachment.py -> build/lib/odoo/addons/hr_expense/models
copying odoo/addons/hr_expense/models/account_move_line.py -> build/lib/odoo/addons/hr_expense/models
copying odoo/addons/hr_expense/models/hr_employee.py -> build/lib/odoo/addons/hr_expense/models
copying odoo/addons/hr_expense/models/account_move.py -> build/lib/odoo/addons/hr_expense/models
copying odoo/addons/hr_expense/models/product_template.py -> build/lib/odoo/addons/hr_expense/models
copying odoo/addons/hr_expense/models/hr_department.py -> build/lib/odoo/addons/hr_expense/models
copying odoo/addons/hr_expense/models/hr_expense.py -> build/lib/odoo/addons/hr_expense/models
copying odoo</pre>
</details>
<details>
<summary>open-web-calendar</summary>
<pre>open_web_calendar/test/test_issue_468_encryption.py .................... [ 46%]
......                                                                   [ 49%]
open_web_calendar/test/test_issue_631_escape_raw_content.py ............ [ 55%]
......                                                                   [ 58%]
open_web_calendar/test/test_issue_680_show_attendees_and_organizer.py .. [ 59%]
..s                                                                      [ 60%]
open_web_calendar/test/test_issue_71_add_javascript.py ....              [ 62%]
open_web_calendar/test/test_special_timezones.py .........               [ 66%]
open_web_calendar/test/test_specification.py .............               [ 72%]
open_web_calendar/test/test_translations.py ............................ [ 86%]
.....                                                                    [ 88%]
open_web_calendar/test/test_url_cache.py ....                            [ 90%]
open_web_calendar/test/test_url_capability.py ............               [ 96%]
open_web_calendar/test/test_version.py ........                          [100%]

======================== 208 passed, 3 skipped in 4.47s ========================
Finished executing pytestCheckPhase
Running phase: pytestcachePhase
Running phase: pytestRemoveBytecodePhase
Connection to build-box.nix-community.org closed by remote host.</pre>
</details>
<details>
<summary>python312Packages.html-sanitizer</summary>
<pre>                    "something    <br>somethingelse    ",
                ),
            ],
            sanitizer=sanitizer,
        )

html_sanitizer/tests.py:424: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
html_sanitizer/tests.py:15: in run_tests
    self.assertEqual(
E   AssertionError: 'Hell[31 chars]o. This is a tabled line.Hello.This is beginning of the end.\n' != 'Hell[31 chars]o. This is a tabled line.Hello.This is beginning of the end.\r'
E     Hello This is a paragraph. 
E   -   Hello. This is a tabled line.Hello.This is beginning of the end.
E   +   Hello. This is a tabled line.Hello.This is beginning of the end.
E   ?                                                                   +
E   - 
E    : b"Cleaning 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\r', expected 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\r' but got 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\n'"
=========================== short test summary info ============================
FAILED html_sanitizer/tests.py::SanitizerTestCase::test_keep_typographic_whitespace - AssertionError: 'Hell[31 chars]o. This is a tabled line.Hello.This is begin...
================== 1 failed, 33 passed, 2 deselected in 0.08s ==================</pre>
</details>
<details>
<summary>python312Packages.trafilatura</summary>
<pre>Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: trafilatura
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
pytest flags: -m pytest -k not\ \(test_cli_pipeline\)\ and\ not\ \(test_crawl_page\)\ and\ not\ \(test_download\)\ and\ not\ \(test_feeds_helpers\)\ and\ not\ \(test_fetch\)\ and\ not\ \(test_input_type\)\ and\ not\ \(test_is_live_page\)\ and\ not\ \(test_meta_redirections\)\ and\ not\ \(test_probing\)\ and\ not\ \(test_queue\)\ and\ not\ \(test_redirection\)\ and\ not\ \(test_whole\)
============================= test session starts ==============================
platform linux -- Python 3.12.11, pytest-8.3.5, pluggy-1.6.0
rootdir: /build/trafilatura-2.0.0
configfile: pyproject.toml
testpaths: tests/*test*.py
collecting ... 
collecting 13 items                                                            
collecting 81 items                                                            
collected 89 items / 12 deselected / 77 selected                               

tests/baseline_tests.py ..                                               [  2%]
tests/cli_tests.py .</pre>
</details>
<details>
<summary>tandoor-recipes</summary>
<pre>Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/ad8w0p0kzi9hrdmp58xdrr27zaswnkxq-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/yarn.lock"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/qrcp1ssyxivaammr6k20kshib4m6b6pk-pytest-xdist.patch
patching file pytest.ini
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>open-web-calendar</summary>
<pre>shrinking RPATHs of ELF executables and libraries in /nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49
checking for references to /build/ in /nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49...
patching script interpreter paths in /nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49
/nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49/lib/python3.13/site-packages/open_web_calendar/__main__.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/avgafrfq80hx92p98m60wh2khmnmxrzc-python3-3.13.4/bin/python3"
/nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49/lib/python3.13/site-packages/open_web_calendar/app.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/avgafrfq80hx92p98m60wh2khmnmxrzc-python3-3.13.4/bin/python3"
stripping (with command strip and flags -S -p) in  /nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49/lib /nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49/bin
shrinking RPATHs of ELF executables and libraries in /nix/store/hgkkag9m9i15x310207zxmf9csqf1rmk-open-web-calendar-1.49-dist
checking for references to /build/ in /nix/store/hgkkag9m9i15x310207zxmf9csqf1rmk-open-web-calendar-1.49-dist...
patching script interpreter paths in /nix/store/hgkkag9m9i15x310207zxmf9csqf1rmk-open-web-calendar-1.49-dist
Rewriting #!/nix/store/avgafrfq80hx92p98m60wh2khmnmxrzc-python3-3.13.4/bin/python3.13 to #!/nix/store/avgafrfq80hx92p98m60wh2khmnmxrzc-python3-3.13.4
wrapping `/nix/store/p13kw60d42qx1gd0arwqs59jzkh5b3b8-open-web-calendar-1.49/bin/open-web-calendar'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
Running phase: pythonRemoveBinBytecodePhase
Running phase: pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: open_web_calendar.app</pre>
</details>
<details>
<summary>python312Packages.html-sanitizer</summary>
<pre>                    "something    <br>somethingelse    ",
                ),
            ],
            sanitizer=sanitizer,
        )

html_sanitizer/tests.py:424: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
html_sanitizer/tests.py:15: in run_tests
    self.assertEqual(
E   AssertionError: 'Hell[31 chars]o. This is a tabled line.Hello.This is beginning of the end.\n' != 'Hell[31 chars]o. This is a tabled line.Hello.This is beginning of the end.\r'
E     Hello This is a paragraph. 
E   -   Hello. This is a tabled line.Hello.This is beginning of the end.
E   +   Hello. This is a tabled line.Hello.This is beginning of the end.
E   ?                                                                   +
E   - 
E    : b"Cleaning 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\r', expected 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\r' but got 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\n'"
=========================== short test summary info ============================
FAILED html_sanitizer/tests.py::SanitizerTestCase::test_keep_typographic_whitespace - AssertionError: 'Hell[31 chars]o. This is a tabled line.Hello.This is begin...
================== 1 failed, 33 passed, 2 deselected in 0.13s ==================</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.html-sanitizer</summary>
<pre>                    "something    <br>somethingelse    ",
                ),
            ],
            sanitizer=sanitizer,
        )

html_sanitizer/tests.py:424: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
html_sanitizer/tests.py:15: in run_tests
    self.assertEqual(
E   AssertionError: 'Hell[31 chars]o. This is a tabled line.Hello.This is beginning of the end.\n' != 'Hell[31 chars]o. This is a tabled line.Hello.This is beginning of the end.\r'
E     Hello This is a paragraph. 
E   -   Hello. This is a tabled line.Hello.This is beginning of the end.
E   +   Hello. This is a tabled line.Hello.This is beginning of the end.
E   ?                                                                   +
E   - 
E    : b"Cleaning 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\r', expected 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\r' but got 'Hello This is a paragraph. \\n\\tHello. This is a tabled line.Hello.This is beginning of the end.\\n'"
=========================== short test summary info ============================
FAILED html_sanitizer/tests.py::SanitizerTestCase::test_keep_typographic_whitespace - AssertionError: 'Hell[31 chars]o. This is a tabled line.Hello.This is begin...
================== 1 failed, 33 passed, 2 deselected in 0.06s ==================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/html-sanitizer/default.nix
+++ b/pkgs/development/python-modules/html-sanitizer/default.nix
@@ -2,20 +2,23 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
+
+  # build-system
   hatchling,
+
+  # dependencies
   lxml,
   lxml-html-clean,
   beautifulsoup4,
+
+  # tests
   pytestCheckHook,
-  pythonOlder,
 }:
 
 buildPythonPackage rec {
   pname = "html-sanitizer";
   version = "2.4.4";
   pyproject = true;
-
-  disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "matthiask";
@@ -24,9 +27,9 @@ buildPythonPackage rec {
     hash = "sha256-6OWFLsuefeDzQ1uHnLmboKDgrbY/xJCwqsSQlDaJlRs=";
   };
 
-  nativeBuildInputs = [ hatchling ];
+  build-system = [ hatchling ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     lxml
     lxml-html-clean
     beautifulsoup4
@@ -40,15 +43,19 @@ buildPythonPackage rec {
     # Tests are sensitive to output
     "test_billion_laughs"
     "test_10_broken_html"
+
+    # Mismatch snapshot (AssertionError)
+    # https://github.com/matthiask/html-sanitizer/issues/53
+    "test_keep_typographic_whitespace"
   ];
 
   pythonImportsCheck = [ "html_sanitizer" ];
 
-  meta = with lib; {
+  meta = {
     description = "Allowlist-based and very opinionated HTML sanitizer";
     homepage = "https://github.com/matthiask/html-sanitizer";
     changelog = "https://github.com/matthiask/html-sanitizer/blob/${version}/CHANGELOG.rst";
-    license = with licenses; [ bsd3 ];
-    maintainers = with maintainers; [ fab ];
+    license = with lib.licenses; [ bsd3 ];
+    maintainers = with lib.maintainers; [ fab ];
   };
 }

--- a/pkgs/development/python-modules/lxml-html-clean/default.nix
+++ b/pkgs/development/python-modules/lxml-html-clean/default.nix
@@ -2,10 +2,9 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
+  setuptools,
   lxml,
   unittestCheckHook,
-  pythonOlder,
-  setuptools,
 }:
 
 buildPythonPackage rec {
@@ -13,14 +12,28 @@ buildPythonPackage rec {
   version = "0.4.2";
   pyproject = true;
 
-  disabled = pythonOlder "3.7";
-
   src = fetchFromGitHub {
     owner = "fedora-python";
     repo = "lxml_html_clean";
     tag = version;
     hash = "sha256-KGUFRbcaeDcX2jyoyyZMZsVTbN+h8uy+ugcritkZe38=";
   };
+
+  # Disable failing snapshot tests (AssertionError)
+  # https://github.com/fedora-python/lxml_html_clean/issues/24
+  # As this derivation must use unittestCheckHook, we cannot use disabledTests
+  postPatch = ''
+    substituteInPlace tests/test_clean.py \
+      --replace-fail \
+        "test_host_whitelist_valid" \
+        "DISABLED_test_host_whitelist_valid" \
+      --replace-fail \
+        "test_host_whitelist_invalid" \
+        "DISABLED_test_host_whitelist_invalid" \
+      --replace-fail \
+        "test_host_whitelist_sneaky_userinfo" \
+        "DISABLED_test_host_whitelist_sneaky_userinfo"
+  '';
 
   build-system = [ setuptools ];
 
@@ -30,11 +43,11 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "lxml_html_clean" ];
 
-  meta = with lib; {
+  meta = {
     description = "Separate project for HTML cleaning functionalities copied from lxml.html.clean";
     homepage = "https://github.com/fedora-python/lxml_html_clean/";
     changelog = "https://github.com/fedora-python/lxml_html_clean/blob/${version}/CHANGES.rst";
-    license = licenses.bsd3;
-    maintainers = with maintainers; [ fab ];
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ fab ];
   };
 }

--- a/pkgs/development/python-modules/trafilatura/default.nix
+++ b/pkgs/development/python-modules/trafilatura/default.nix
@@ -1,17 +1,22 @@
 {
   lib,
   buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  setuptools,
+
+  # dependencies
   certifi,
   charset-normalizer,
   courlan,
-  fetchPypi,
   htmldate,
   justext,
   lxml,
-  pytestCheckHook,
-  pythonOlder,
-  setuptools,
   urllib3,
+
+  # tests
+  pytestCheckHook,
 }:
 
 buildPythonPackage rec {
@@ -19,11 +24,11 @@ buildPythonPackage rec {
   version = "2.0.0";
   pyproject = true;
 
-  disabled = pythonOlder "3.9";
-
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-zrcJSm7Ml+cv6nPH26NnFMXFtXe2Rw5FINyok3BtYkc=";
+  src = fetchFromGitHub {
+    owner = "adbar";
+    repo = "trafilatura";
+    tag = "v${version}";
+    hash = "sha256-Cf1W3JEGSMkVmRZVTXYsXzZK/Nt/aDG890Sf0/0OZAA=";
   };
 
   postPatch = ''
@@ -48,6 +53,15 @@ buildPythonPackage rec {
   nativeCheckInputs = [ pytestCheckHook ];
 
   disabledTests = [
+    # TypeError: argument of type 'NoneType' is not iterable
+    # https://github.com/adbar/trafilatura/issues/805
+    "test_external"
+    "test_extract"
+
+    # AttributeError: 'NoneType' object has no attribute 'find'
+    # https://github.com/adbar/trafilatura/issues/805
+    "test_table_processing"
+
     # Disable tests that require an internet connection
     "test_cli_pipeline"
     "test_crawl_page"

